### PR TITLE
Added table cell component fixed text-wrapping issue.

### LINKED ISSUE #2051: Same component for the cells in plugin and job tabs
There are several fields that shows a string (ex: name, user, playbook_execute, etc...) and all of them have the same UX (allow to copy text with a single click and truncate with `...` in case the text is too long for the column size). Have a component for all of them will improve the code and the maintenance.

### PR DESCRIPTION
**Pull request by aryan-bhokare**
closes #2051

# Description

Please include a summary of the change and link to the related issue.
Added a property to wrap the text in the cell so that is does not go out of bound.
Made a component for a `TableCell` component to improvise reusability.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [x] The pull request is for the branch `develop`
- [ ] A new plugin (analyzer, connector, visualizer, playbook, pivot or ingestor) was added or changed, in which case:
    - [ ] I strictly followed the documentation ["How to create a Plugin"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-add-a-new-plugin)
    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Advanced-Usage.md) was updated (in case the plugin provides additional optional configuration).
    - [ ] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [ ] I have dumped the configuration from Django Admin using the `dumpplugin` command and added it in the project as a data migration. (["How to share a plugin with the community"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-share-your-plugin-with-the-community))
    - [ ] If a File analyzer was added and it supports a mimetype which is not already supported, you added a sample of that type inside the archive `test_files.zip` and you added the default tests for that mimetype in [test_classes.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_classes.py).
    - [ ] If you created a new analyzer and it is free (does not require API keys), please add it in the `FREE_TO_USE_ANALYZERS` playbook by following [this guide](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-modify-a-plugin).
    - [ ] Check if it could make sense to add that analyzer/connector to other [freely available playbooks](https://intelowl.readthedocs.io/en/develop/Usage.html#list-of-pre-built-playbooks).
    - [ ] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results.
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] I have added tests for the feature/bug I solved (see `tests` folder). All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Contribute.md)).
- [x] If the GUI has been modified:
    - [x] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.

Screenshots:

Screenshot from 2024-02-27 00-06-09

Screenshot from 2024-02-27 00-05-42

Screenshot from 2024-02-27 00-05-08

### CODE REVIEW

**Comment by aryan-bhokare** [comment]
@drosetti PTAL .

Also I would like to component this 

```
      <ul className="d-flex flex-column align-items-start text-left text-break">
        {value?.sort().map((val) => (
          <li className="mb-2 pb-2" key={val}>
            {val}
          </li>
        ))}
      </ul>
```

As it's being reused a lot.

**Comment by drosetti** [comment]
* There is a bug in the visualizer section (attached file)
Screenshot from 2024-02-27 10-32-26

* I also noticed a bug in the ingestor - playbook_to_execute: in case of long text it neither truncate nor break the playbook name.
 * Talking about your first screenshot: Do you think is possibile to have the 3 dots and the copyToClipboard in the elements instead of break the text ?
* In order to answer to your question: yes, a component for the list it would be great!

**Comment by aryan-bhokare** [comment]
Screenshot from 2024-02-28 00-38-59

Screenshot from 2024-02-28 00-37-40

Screenshot from 2024-02-27 23-56-12

**Comment by drosetti** [comment]
Thanks for the changes and the work you done! I'd like to talk with @mlodic about the UX. I know you did the changes I requested, but i'd like to have his opinion about it

**Comment by aryan-bhokare** [comment]
> Thanks for the changes and the work you done! I'd like to talk with @mlodic about the UX. I know you did the changes I requested, but i'd like to have his opinion about it

Sure, Thanks!

**Comment by drosetti** [comment]
@aryan-bhokare I talked with Matteo and we would like these changes:

* plugins -> analyzer -> supported types: ok the truncate, but it's too early, there is lots of free space available (see screenshot)
Screenshot from 2024-03-04 17-30-40
* plugins -> visualizer -> playbook connected to: truncate too early
* plugins -> ingestor -> playbook_to_execute: truncate with the `...`
* plugins -> playbooks -> analyzer/connector/...: truncate too early
* jobs -> playbook executued: add truncate and copy feature

Thanks!

**Comment by aryan-bhokare** [comment]
> @aryan-bhokare I talked with Matteo and we would like these changes:
> 
> * plugins -> analyzer -> supported types: ok the truncate, but it's too early, there is lots of free space available (see screenshot)
>   Screenshot from 2024-03-04 17-30-40
> * plugins -> visualizer -> playbook connected to: truncate too early
> * plugins -> ingestor -> playbook_to_execute: truncate with the `...`
> * plugins -> playbooks -> analyzer/connector/...: truncate too early
> * jobs -> playbook executued: add truncate and copy feature
> 
> Thanks!

I'll have to truncate dynammically then.
let me check if we have something

**Comment by drosetti** [comment]
I think the best way to truncate text is to create the div of the correct dimension for the ul elements and then use the truncate from bootstrap. Idk if this can help you, if you have some doubt let me know :)

**Comment by aryan-bhokare** [comment]
@drosetti I have made all the requested changes.

**Review by drosetti — CHANGES_REQUESTED** [review]
My apologizes to request you to "revert" something about the UX, but can you restore the supported_types in plugin -> analyzers and the elements in the columns of playbooks -> analyzers/connectors/... to break the text instead of truncate ? Even if I think it's more beautiful the truncate version, the text break show all the type or plugin name and it could be more useful.

Thanks for your work!

Edit: I think it should be quite easy, you structured well the components so it should be just a flag.

File: frontend/src/components/common/TableCell.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          <span
+            className="d-block text-break pb-2"
+            id={`table-user-${job ? job.id : value}`}
+            key={`table-user-${job ? job.id : value}`}
+          >
+            {value}
+          </span>
+        </div>
+      )}
+    </div>
```
[the remark this answers is not in the document]
Comment (aryan-bhokare) [reply frontend/src/components/common/TableCell.jsx:40]: ohh yes... didn't think of it thanks!

File: frontend/src/components/jobs/table/jobTableColumns.jsx [hunk 2]
```diff
@@ -230,9 +208,18 @@ export const jobTableColumns = [
...
+              text={
+                playbooks?.find(
+                  (playbook) => playbook.id === job.playbook_to_execute,
+                )?.name || "Custom Analysis"
+              }
+            >
+              {playbooks?.find(
+                (playbook) => playbook.id === job.playbook_to_execute,
+              )?.name || "Custom Analysis"}
+            </CopyToClipboardButton>
```
[the remark this answers is not in the document]
Comment (aryan-bhokare) [reply frontend/src/components/jobs/table/jobTableColumns.jsx:222]: I don't see this being used multiple times it's only specific to playbook to execute.

File: frontend/src/components/common/TableCell.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aryan-bhokare) [reply frontend/src/components/common/TableCell.jsx]: done

File: frontend/src/components/jobs/table/jobTableColumns.jsx [hunk 3]
```diff
@@ -69,15 +70,7 @@ export const jobTableColumns = [
...
-      <CopyToClipboardButton
-        showOnHover
-        id={`table-user-${job?.id}`}
-        key={`table-user-${job?.id}`}
-        text={value}
-        className="d-block text-truncate"
-      >
-        {value}
-      </CopyToClipboardButton>
+      <TableCell job={job} isCopyToClipboard value={value} />
```
[the remark this answers is not in the document]
Comment (aryan-bhokare) [reply frontend/src/components/jobs/table/jobTableColumns.jsx]: done

File: frontend/src/components/plugins/types/utils.jsx [hunk 4]
```diff
@@ -274,9 +280,17 @@ export function PlaybooksCollapse({ pluginList, pluginType_ }) {
...
-            <li key={pluginName}>{pluginName}</li>
+            <li className="pb-2" key={pluginName}>
+              <CopyToClipboardButton
+                showOnHover
+                text={pluginName}
+                className="d-block text-truncate"
+              >
+                {truncateText(pluginName, 12)}
+              </CopyToClipboardButton>
+            </li>
```
[the remark this answers is not in the document]
Comment (aryan-bhokare) [reply frontend/src/components/plugins/types/utils.jsx]: done

File: frontend/src/components/plugins/types/pluginTableColumns.jsx [hunk 5]
```diff
@@ -175,13 +168,7 @@ export const analyzersTableColumns = [
...
       return supported;
     },
-    Cell: ({ value }) => (
-      <ul className="d-flex flex-column align-items-start">
-        {value?.sort().map((val) => (
-          <li key={val}>{val}</li>
-        ))}
-      </ul>
-    ),
+    Cell: ({ value }) => <TableCellList value={value} size={25} />,
```
[the remark this answers is not in the document]
Comment (aryan-bhokare) [reply frontend/src/components/plugins/types/pluginTableColumns.jsx]: done

File: frontend/src/components/jobs/table/jobTableColumns.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (aryan-bhokare) [reply frontend/src/components/jobs/table/jobTableColumns.jsx:222]: I don't this this login being repeated. It's specific to playbook to execute only

[the remark this answers is not in the document]
Comment (drosetti) [reply frontend/src/components/jobs/table/jobTableColumns.jsx:222]: I'm talking about:
```
playbooks?.find(
  (playbook) => playbook.id === job.playbook_to_execute,
)?.name || "Custom Analysis"
```
It's not used in other files, but in this one is used twice

[the remark this answers is not in the document]
Comment (aryan-bhokare) [reply frontend/src/components/jobs/table/jobTableColumns.jsx]: done

**Comment by drosetti** [comment]
@aryan-bhokare please when you think the code is ready request me the re-review. In this way I'll know if I have to look at the code or I'll think you are still working

**Comment by aryan-bhokare** [comment]
> @aryan-bhokare please when you think the code is ready request me the re-review. In this way I'll know if I have to look at the code or I'll think you are still working

ohh sorry can you review it @drosetti ?

**Review by drosetti — CHANGES_REQUESTED** [review]
Just last request, can you change different keys ? I noticed some errors in the console:
Screenshot from 2024-03-19 11-03-21

**Review by drosetti — APPROVED** [review]

**Comment by drosetti** [comment]
Top! Thanks for your contribution :)

**Comment by aryan-bhokare** [comment]
> Top! Thanks for your contribution :)

Happy to contribute 😁.

**Pull request merged**